### ISSUE TITLE
Crash on back navigation to map activity (Android)

### ISSUE THREAD

**Issue by pascal-hauser** [mapbox/mapbox-gl-native#12440]
We have the map with some location markers. On tapping the info window we start another activity. Once the 2nd activity finishes, the app crashs instead of showing the map again.

logcat shows
....
I/Process: Sending signal. PID: 7952 SIG: 9
W/InputDispatcher: channel '5543de <packagename>/<...MapActivity> (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
E/InputDispatcher: channel '5543de <packagename>/<...MapActivity> (server)' ~ Channel is unrecoverably broken and will be disposed!
W/InputDispatcher: channel '39291a9 <packagename>/<...OtherActivity> (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
E/InputDispatcher: channel '39291a9 <packagename>/<...OtherActivity> (server)' ~ Channel is unrecoverably broken and will be disposed!
E/SurfaceFlinger: Failed to find layer (SurfaceView - <packagename>/<...MapActivity>#0) in layer parent (no-parent).
    Failed to find layer (Background for - SurfaceView - <packagename>/<...MapActivity>#0) in layer parent (no-parent).
I/ActivityManager: Process <packagename> (pid 7952) has died: fore TOP 
W/ActivityManager: Scheduling restart of crashed service <packagename>/com.mapbox.android.telemetry.TelemetryService in 28128ms
W/zygote64: kill(-7952, 9) failed: No such process
I/zygote64: Successfully killed process cgroup uid 10175 pid 7952 in 0ms
I/WindowManager: WIN DEATH: Window{39291a9 u0 <packagename>/<...OtherActivity>}
W/InputDispatcher: Attempted to unregister already unregistered input channel '39291a9 <packagename>/<...OtherActivity> (server)'
I/WindowManager: WIN DEATH: Window{5543de u0 <packagename>/<...MapActivity>}
W/InputDispatcher: Attempted to unregister already unregistered input channel '5543de <packagename>/<...MapActivity> (server)'
W/ActivityManager: Force removing ActivityRecord{842a93 u0 <packagename>/<...MapActivity> t3927}: app died, no saved state
....
and sometimes also this is logged
....
--------- beginning of crash
07-19 16:12:24.185  1823  1823 E AndroidRuntime: FATAL EXCEPTION: main
07-19 16:12:24.185  1823  1823 E AndroidRuntime: Process: <ourapppackagename>, PID: 1823
07-19 16:12:24.185  1823  1823 E AndroidRuntime: java.lang.RuntimeException: Unable to stop service com.mapbox.android.telemetry.TelemetryService@13c1cc: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.SharedPreferences android.content.Context.getSharedPreferences(java.lang.String, int)' on a null object reference
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.app.ActivityThread.handleStopService(ActivityThread.java:3522)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.app.ActivityThread.-wrap26(Unknown Source:0)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1697)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6494)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
07-19 16:12:24.185  1823  1823 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.SharedPreferences android.content.Context.getSharedPreferences(java.lang.String, int)' on a null object reference
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.mapbox.android.telemetry.TelemetryUtils.obtainSharedPreferences(TelemetryUtils.java:191)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.mapbox.android.telemetry.TelemetryLocationEnabler.updateLocationPreferences(TelemetryLocationEnabler.java:61)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.mapbox.android.telemetry.TelemetryLocationEnabler.updateTelemetryLocationState(TelemetryLocationEnabler.java:40)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.mapbox.android.telemetry.TelemetryService.disableTelemetryLocationState(TelemetryService.java:243)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at com.mapbox.android.telemetry.TelemetryService.onDestroy(TelemetryService.java:65)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        at android.app.ActivityThread.handleStopService(ActivityThread.java:3504)
07-19 16:12:24.185  1823  1823 E AndroidRuntime:        ... 8 more
....

With an older version ('com.mapbox.mapboxsdk:mapbox-android-sdk:6.1.3') this doesn't happen and the map shows again as expected.

### Steps to reproduce

 1. Start another activity on info window tap
 2. Finish the other activity
 3. Crash
(probably some other things required to get this behavior

### Expected behavior
Map shown again.

### Actual behavior
Crash (without showing an app crash dialog).

### Configuration

**Android versions:**
8.1
**Device models:** 
Pixel 2 XL
**Mapbox SDK versions:** 
Affected: 6.3.0
Not affected: 6.1.3

**Comment by pascal-hauser** [comment]
Meanwhile I also found issue #7527 that could be related to this.

**Comment by EvangelosG** [comment]
@pascal-hauser This happens to me occasionally when returning to an activity w/ mapbox. The worst part about mapbox crashes is the nasty ones like these aren't caught by crashlytics (signal 11, signal 7, etc...), your app just dies. Look at the # of issues on this repo (currently 682), then look at the resolved ones, almost all end with no change/denial/request(s) for more info. This SDK's repo is a prime example of terrible SDK management. You are going to be chasing a faster than light dragon attempting to use this SDK. I just switched to google maps (34 current repo reported issues), it's free for **mobile maps** and their watermark isn't clickable!...

**Comment by pascal-hauser** [comment]
@tobrun No, we are not using any plugins.
@AndoJava We are still investigating. Unfortunately the crash doesn't give any useful information in logcat and so far we were not able to compile the sdk to make use of a tumbstone dump with the NDK tools.
In general I think the exception posted above should not happen, but it is only a side effect. The app is already dying and that's why the mapbox service is crashing too.
Still unclear is why the app is dying. It only happens on one activity navigation, one that has Unity player included. Current guess is, and this is only a guess, that  there is a conflict around access through GL surface. Any ideas or hints to narrow that down?
@EvangelosG Thank you for your feedback. Yes we found too that SIGSEGV and SIGABRT are very hard to narrow down. The issue with the issues we saw too, makes one wondering. We have google maps for other apps, but fore some we have mapbox and switching to google maps is not an option. So we still hope on getting this fixed, or find a workaround.
